lisdf/planner_output/command.py

```diff
@@ -0,0 +1,238 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import ClassVar, Dict, List, Optional, Type
+
+import numpy as np
+
+from lisdf.planner_output.common import OutputElement
+
+
+class Command(OutputElement, ABC):
+    """
+    Hack based on Jiayuan's code for enforcing subclasses to define a type which is
+    then used for serialization.
+
+    When inherit from this class, child classes should pass type="XXX"
+    as a keyword argument. This will register a new Command type
+    in this class which we use when deserializing from a JSON dict.
+    """
+
+    # TODO: figure out a clean way to move the 'label' field here
```

Comment:
Why can't we have the label field here? Not super familiar with abstractbaseclasses

Reply:
We want label to have a default value of None. Inheritance with dataclasses with default value fields doesn't work without lots of hacks.